python/guide/module/autograd.md

```diff
@@ -1,178 +1,260 @@
-automatically calculating derivatives. And while most other libraries require
-that we compile a symbolic graph to take automatic derivatives, `autograd`
-allows you to take derivatives while writing  ordinary imperative code. Every
-time you make pass through your model, `autograd` builds a graph on the fly,
-through which it can immediately backpropagate gradients.
+### Why do we need to calculate gradients?
+
+Gradients are fundamental to the process of training neural networks, and tell us how to change the parameters of the network to improve its performance.
+
+<p align="center"><img src="./imgs/autograd_gradient.png" alt="drawing" width="400"/></p>
+
+Long Answer:
+
+Under the hood, neural networks are composed of operators (e.g. convolutions) which use parameters (e.g. weights in convolution kernels) for their computation, and it's our job to find the optimal values for these parameters. Gradients lead us to the solution!
+
+Gradients tell us what happens to a variable (increases/decrease and by how much), when we change another variable that it depends on. What we're interested in is the effect of changing each parameter on the performance of the network. We usually define performance using a loss metric that we try to minimize, i.e. a metric that tells us how bad the predictions of a network are given ground truth. As an example, for regression we might try to minimize the L2 loss (euclidean distance) between our predictions and true values, and for classification we minimize the cross-entropy loss. 
+
+Assuming we've calculated the gradient of each parameter with respect to the loss (details in next section), we can then use an optimizer such as Stochastic Gradient Descent to shift the parameters slightly in the *opposite direction* of the gradient. See [Optimizers](http://mxnet.incubator.apache.org/api/python/optimization/optimization.html) for more information on these methods. We repeat the process of calculating gradients and updating parameters over and over again, until the parameters of the network start to stabilize and converge to a good solution. 
+
+## How do we calculate gradients?
+
+We differentiate. MXNet Gluon uses Reverse Mode Automatic Differentiation (`autograd`) to backpropagate gradients from the loss metric to the network parameters.
+
+<p align="center">
+    <img src="./imgs/autograd_forward_backward.png" alt="drawing" width="400"/>
+</p>
+
+Long Answer:
+
+One option would be to get out our calculus books and work out the gradients by hand. Who wants to do this though? It's time consuming and error prone for starters. Another option is Symbolic Differentiation, which calculates the formulas for each gradient, but this quickly leads to incredibly long formulas as networks get deeper and operators get more complex. We could use finite differencing, and try slight differences on each parameter and see how the loss metric responds, but this is computationally expensive and has poor numerical precision.
```

Comment:
no need for capitalization of symbolic diff. but link to an article

Reply:
I would qualify the poor numerical precision or provide a ref. I would guess, it is difficult to maintain precision through many chain rules.

A huge advantage of backpropagation is dynamic programming.

Reply:
done.